### ISSUE TITLE
training my own data set with ']' output predict

### ISSUE THREAD

**Issue by ziyan-wyq** [thtrieu/darkflow#300]
hi I tried to train my own data set using cfg/tiny-yolo-voc.cfg tiny-yolo-voc.weight.
I have changed the class to 2, lable.txt, and the last conv filter to 35(5*(2+5))
the training script is as follow:
./flow --model cfg/tiny-yolo-voc-1.cfg --load bin/tiny-yolo-voc.weights --train --annotation train/Annotations --dataset train/JPEGImages --gpu 1.0 --batch 30 --epoch 100
after trained the loss is under 2.0
the testing script is:
/flow --imgdir test_dir/ --model cfg/tiny-yolo-voc.cfg --load -1 --json
the result is ']'
why is it ?
the sample under test data is just a training data.

**Comment by thtrieu** [comment]
First, a single ']' means there is no object detected.

Second, a single ']' is incorrect JSON format, and someone should fix this. Using text concatenation is quite unreliable to produce JSON. We should use `json.dumps` instead

https://github.com/thtrieu/darkflow/blob/master/darkflow/net/yolo/predict.py#L119

**Comment by vicwer** [comment]
1. the same problem occors to me.
I have change the class to 'person' labels.txt, and the the conv filter to 30.
the training script is:  flow --model cfg/yolo-voc-0623.cfg --train --dataset ~/kgduan/yolo/darkflow/VOCdevkit/VOC2007/JPEGImages/ --annotation ~/kgduan/yolo/darkflow/VOCdevkit/VOC2007/Annotations/ --gpu 1.0
the testing script is: flow --model cfg/yolo-voc-0623.cfg --load 6000 --json
the result is ']'
why?
2. I can`t resume to train.
the scripyis: flow --train --model cfg/yolo-voc-0623.cfg --load -1
but mistake occours:
Enter training ...
Error: Annotation directory not found ../pascal/VOCdevkit/ANN/ .
why...
3.when I change learning rate to 0.001 or 0.0001 (and batch size to 64 or 32)in yolo-voc-0623.cfg,it always is:
Training statistics: 
	Learning rate : 1e-05
	Batch size    : 16
	Epoch number  : 1000
	Backup every  : 2000
I don`t know why...
can you help me, please...

**Comment by kiki9014** [comment]
I have same problem.
I followed README.md but test image didn't generate any box.
In debug mode, I found that findboxes function returns [].
I think box_constructor in cy_yolo2_findboxes does not work at all.

**Comment by Matsam195** [comment]
@thtrieu : What kind of json output should we expect from a prediction with no bounding boxes? Just an empty list "[]"?

@vicwer : If you want to modify learning rate, batch size, epoch number, etc., you should use arguments in command prompt instead of changing the configuration file. Your command becomes:
`flow --train --model cfg/yolo-voc-0623.cfg --load -1 --lr 0.0001 --batch 32`
This is because default arguments are set to 1e-05, 16 and 1000 respectively for learning rate, batch size and epoch number. So basically I'm not sure these variables in the configuration file have any use.

**Comment by abagshaw** [comment]
@Matsam195 I think an empty list `[]` would be best for a JSON result with no bounding boxes. When I have a chance I can change the script over to `json.dumps` as @thtrieu suggested to fix this.

In regards to the learning rate, epoch number, and other training variables being set in the `.cfg` I believe these are used for training with the original https://github.com/pjreddie/darknet but they are ignored when training with darkflow.

**Comment by abagshaw** [comment]
The invalid JSON formatting should be fixed as of https://github.com/thtrieu/darkflow/pull/315

**Comment by Heidisnaps** [comment]
@abagshaw How did yo fix it? 
I have same issue.
Could you give me a code example?
And How did you set the options when training the model?

in my case..
`flow --train --model cfg/tiny-yolo-voc-new3.cfg --dataset "~/VOCdevkit/VOC2007/JPEGImages" --annotation "~/VOCdevkit/VOC2007/Annotations" --trainer adam --gpu 0.9 --save 150`

but it also result is '[]'
